barretenberg/cpp/src/barretenberg/vm2/tracegen/lib/phase_spec.cpp

```diff
@@ -0,0 +1,83 @@
+#include "barretenberg/vm2/tracegen/lib/phase_spec.hpp"
+#include "barretenberg/vm2/common/aztec_constants.hpp"
+
+#include <array>
+#include <cstdint>
+#include <unordered_map>
+#include <vector>
+
+namespace bb::avm2::tracegen {
+
+namespace {
+// TODO: check back after public inputs are done
+const std::unordered_map<TransactionPhase, TxPhaseOffsetsTable::Offsets> PHASE_PUB_INPUTS_OFFSETS = { {
+    { TransactionPhase::SETUP,
+      {
+          /*read_offset=*/AVM_PUBLIC_INPUTS_PUBLIC_SETUP_CALL_REQUESTS_ROW_IDX,
```

Comment:
Given this is a structure, I think you can do `.read_offset=`, etc instead of comments, which is actually safer because you don't depend on order